specutils/spectra/spectrum1d.py

```diff
@@ -265,7 +265,7 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
         elif wcs is None:
             # If no spectral axis or wcs information is provided, initialize
             # with an empty gwcs based on the flux.
-            size = len(flux) if not flux.isscalar else 1
+            size = flux.shape[-1] if not flux.isscalar else 1
```

Comment:
Wondering if the `isscalar` check is even necessary after the `flux = u.Quantity([flux])` further up.
But then there is an alternative hook where `flux = kwargs.pop('data')` might be slipping through; not sure if that is by design?

Reply:
🤷